pywren/jobrunner.py

```diff
@@ -1,33 +1,88 @@
 from __future__ import print_function
-
+import os
 import base64
 import json
 import sys
 import traceback
+import boto3
+
 
 from six.moves import cPickle as pickle
 from tblib import pickling_support
 
 pickling_support.install()
 
+# def s3_url_parse(url):
+#     if url[:5] != "s3://":
+#         raise Exception("improperly formatted s3 url: {}".format(url))
+#     bucket, key = url[5:].split("/", 1)
+#     return bucket, key
+
 def b64str_to_bytes(str_data):
     str_ascii = str_data.encode('ascii')
     byte_data = base64.b64decode(str_ascii)
     return byte_data
 
 try:
-    func_filename = sys.argv[1]
-    data_filename = sys.argv[2]
-    out_filename = sys.argv[3]
+    jobrunner_config_filename = sys.argv[1]
+    out_filename = sys.argv[2]
+
+
     # initial output file in case job fails
     pickle.dump({'result' : None,
                  'success' : False},
                 open(out_filename, 'wb'), -1)
+    jobrunner_config = json.load(open(jobrunner_config_filename,
+                                      'r'))
+
+
+    # FIXME someday switch to storage handler
+    # download the func data into memory
+    s3_client = boto3.client("s3")
+    func_bucket, func_key = jobrunner_config['func_bucket'], jobrunner_config['func_key']
+    data_bucket, data_key = jobrunner_config['data_bucket'], jobrunner_config['data_key']
+
+    data_byte_range = jobrunner_config['data_byte_range']
+    func_obj_stream = s3_client.get_object(Bucket=func_bucket, Key=func_key)
```

Comment:
this is already supported by the storage API right ? is there any reason we can't directly use it ?